quinn-proto/src/endpoint.rs

```diff
@@ -111,8 +111,10 @@ where
     ) -> Option<ConnectionEvent> {
         use EndpointEventInner::*;
         match event.0 {
-            NeedIdentifiers(n) => {
-                return Some(self.send_new_identifiers(ch, n));
+            NeedIdentifiers(now, n) => {
+                if self.connections.get(ch.into()).is_some() {
```

Comment:
Why was this guard introduced?

Reply:
I remember the condition check was added there to prevent connection is draining while the PushNewCid timer alarms. I am moving this check into `connection/mod.rs` before pushing the `NeedIdentifiers` event

Reply:
Ah... i think `is_closed` already covers the `Draining`. We should be good to remove this check